### PR TITLE
docs for source_credentials.json

### PR DESCRIPTION
**Pull request by memsharded**

### CODE REVIEW

File: reference/config_files/source_credentials.rst [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+.. include:: ../../common/experimental_warning.inc
+
+
+When a ``conanfile.py`` recipe downloads some sources from other servers with the ``download()`` or the ``get()`` helpers like:
+
+.. code-block:: python
+
+    def source(self):
+        # Immutable source .zip
+        download(self, f"https://some/server/tarballname-{self.version}.zip", "downloaded.zip")
```
Comment (RubenRBS) [inline reference/config_files/source_credentials.rst:15]: Using the same URL thru the file to show that you won't need to match the whole URL, just the begining

```suggestion
        download(self, f"https://server/that/need/credentials/files/tarballname-{self.version}.zip", "downloaded.zip")
```

File: reference/config_files/source_credentials.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+URL that should match the recipe one. If the recipe URL starts with the given one in the credentials files,
+then the credentials will be injected. If the file provides multiple credentials for multiple URLs, they
+will be evaluated in order until the first match happens. If no match is found, no credentials will be injected.
+
+.. code-block:: json
+
+    {
+        "credentials": [
+            {
+                "url": "http://server/that/need/credentials",
```
Comment (RubenRBS) [inline reference/config_files/source_credentials.rst:34]:
```suggestion
                "url": "https://server/that/need/credentials", 
```

File: reference/config_files/source_credentials.rst [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+If for some reason HTTP-Basic auth with user/password is necessary it can be provided with the ``user`` and
+``password`` fields:
+
+.. code-block:: json
+  
+    {
+        "credentials": [
+            {
+                "url": "http://server/that/need/credentials",
```
Comment (RubenRBS) [inline reference/config_files/source_credentials.rst:51]:
```suggestion
                "url": "https://server/that/need/credentials", 
```

File: reference/config_files/source_credentials.rst [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+level:
+
+.. code-block:: jinja
+    
+    {% set mytk = 'mytoken' %}
+    {
+        "credentials": [
+            {
+                "url": "http://server/that/need/credentials", 
+                "token": "{{mytk}}"
```
Comment (RubenRBS) [inline reference/config_files/source_credentials.rst:70]: As you mentioned `os.getenv`, let's have the example also use it

```suggestion
    {% set mytk = os.getenv('mytoken') %}
    {
        "credentials": [
            {
                "url": "https://server/that/need/credentials", 
                "token": "{{mytk}}"
```

**Review by RubenRBS — APPROVED** [review]
Nice! Some small suggestions

**Review by czoido — APPROVED** [review]

**Pull request merged**